packages/vite/src/node/optimizer/index.ts

```diff
@@ -1210,7 +1212,9 @@ function getConfigHash(environment: Environment): string {
   const { optimizeDeps } = config
   const content = JSON.stringify(
     {
-      mode: process.env.NODE_ENV || config.mode,
+      define: config.keepProcessEnv
```

Comment:
if `config.keepProcessEnv` is `false` then `config.mode` is ignored

this doesn't seem correct to me, is it?

Reply:
```suggestion
      define: !config.keepProcessEnv
```
Good catch! Boolean conditions are always hard 🫠